app/controllers/payment_controller.rb

```diff
@@ -109,4 +111,15 @@ def id_for_response
       { transaction_id: builder.transaction_id }
     end
   end
+
+  def verify_bot
+    action   = 'donate/' + params[:page_id]
+    @captcha = Recaptcha3.new(token: params[:recaptacha_token], action: action)
+
+    unless @captcha.human?
+      msg = @captcha.errors.present? ? @captcha.errors : 'Invalid request'
+      render json: { success: false, message: msg }, status: :unprocessible_entity
+      return false
```

Comment:
Does `return` here add anything?

Reply:
@osahyoun 
It halt further processing. We can see filter chain blocked  message in log.